wrapper/src/main/java/software/amazon/jdbc/plugin/failover/FailoverConnectionPlugin.java

```diff
@@ -726,12 +707,23 @@ protected void invalidateCurrentConnection() {
               originalHost.getPort(),
               originalHost.getRole(),
               HostAvailability.NOT_AVAILABLE));
-      this.pluginService.setAvailability(originalHost.getAliases(), HostAvailability.NOT_AVAILABLE);
     } catch (final SQLException e) {
       LOGGER.fine(() -> Messages.get("Failover.failedToUpdateCurrentHostspecAvailability"));
     }
   }
 
+  protected void invalidateCurrentConnection(final Connection conn) {
+    if (conn == null) {
+      return;
+    }
+
+    final HostSpec originalHost = this.pluginService.getCurrentHostSpec();
+    isInTransaction = this.pluginService.isInTransaction();
+
+    this.pluginService.invalidateAllConnections(originalHost);
+    this.pluginService.setAvailability(originalHost.getAliases(), HostAvailability.NOT_AVAILABLE);
+  }
+
   protected synchronized void pickNewConnection() throws SQLException {
```

Comment:
When failover is processed, we need to invalidate nodes that changed their roles. I haven't found this logic. Do we need it?